### ISSUE TITLE
match development hangs on `Cloning remote git repo...` v0.6.1

### ISSUE THREAD

**Issue by talezion** [fastlane/fastlane#5473]
I get this hanging for ages

**Comment by alexvbush** [comment]
Same here

**Comment by TKBurner** [comment]
@talezion Thanks for reaching out. Have you used match successfully on this setup before? Can you use the full issue template and provide your full input and output? Thanks for working with me 👍

**Comment by talezion** [comment]
@TKBurner Thank you very much for your help, this is the first time I am running match on this mac. Could you please advise how I would produce the full issue template?

Thanks

**Comment by TKBurner** [comment]
Hi @talezion Thanks for following up. Copy/paste the template here, https://github.com/fastlane/fastlane/issues/new

Thanks!

**Comment by KrauseFx** [comment]
When scan hangs when cloning the repo, run it using `--verbose` so that you see the commands that are being executed. Then cancel the process when it hangs and copy & paste the command that does the `git clone`. When you run it manually, you'll see some kind of login prompt or error message. 
Make sure to have read and write access to the repo and have git properly set up on the machine (make sure to use the correct protocol: `https` or `ssh`, depending on your setup)

**Comment by juanpdelat** [comment]
@KrauseFx when cloning the repo manually with the command that match showed using `--verbose`, it asks me for username and password. I am trying this in a new machine (not the one I used to setup match the first time) and I have already added my ssh keys to github.

**Comment by juanpdelat** [comment]
Found my problem, I had setup match to use the https url instead of ssh, changing the matchfile to use the ssh url fixed my problem.

**Comment by KrauseFx** [comment]
Yeah, great to hear you got it resolved @juanpdelat 👍

**Comment by cybertk** [comment]
Same issue here, as the certificates host is not in  **~/.ssh/known_hosts**.

**Comment by TKBurner** [comment]
@cybertk Can you share your full output with the --verbose flag? Thanks!

**Comment by cybertk** [comment]
```
$ match --verbose

+-----------------------+--------------------------------------------------------+
|                            Summary for match 0.6.3                             |
+-----------------------+--------------------------------------------------------+
| verbose               | true                                                   |
| git_url               | git@example.com:cybertk/certificates.git               |
| type                  | development                                            |
| app_identifier        | com.example.app                                        |
| readonly              | true                                                   |
| git_branch            | master                                                 |
| username              | developer.apple@example.com                            |
| keychain_name         | login.keychain                                         |
| force                 | false                                                  |
| shallow_clone         | false                                                  |
| force_for_new_devices | false                                                  |
| skip_docs             | false                                                  |
+-----------------------+--------------------------------------------------------+

INFO [2016-09-14 14:34:43.19]: Cloning remote git repo...
INFO [2016-09-14 14:34:43.19]: $ git clone 'git@example.com:cybertk/certificates.git' '/var/folders/ft/d3401yq51xn4_8yyh18kqfsr0000gq/T/d20160914-56782-e4on20'
INFO [2016-09-14 14:34:43.20]: ▸ Cloning into '/var/folders/ft/d3401yq51xn4_8yyh18kqfsr0000gq/T/d20160914-56782-e4on20'...
INFO [2016-09-14 14:34:43.47]: ▸ The authenticity of host 'example.com (10.0.0.2)' can't be established.
INFO [2016-09-14 14:34:43.47]: ▸ ECDSA key fingerprint is SHA256:mVt+rFhAugxsrNhamGD8nS1En04UOZqNCA1+gOzQOic.
```

**Comment by TKBurner** [comment]
@cybertk So sorry for the delay in getting back to you. Hope you are doing well. What are you using for your git_url? Do you have a github repo setup for it?

Thanks!

**Comment by parleer** [comment]
Latest Match (0.11.1)  doesn't work. Stuck on "Cloning remote git repo..." 

My git repo is in ~/.ssh/known_hosts.

My keychain is unlocked. 

git version 2.11.0


```

INFO [2016-12-14 02:43:48.70]: Cloning remote git repo...
INFO [2016-12-14 02:43:48.70]: $ GIT_TERMINAL_PROMPT=0 git clone 'git@gitlab.mydomain.com:my-group/my-app-certificates.git' '/var/folders/lc/wjt2vfs50vsbd38_n8cp551h0000gn/T/d20161214-59866-xxnfgf'
INFO [2016-12-14 02:43:48.70]: ▸ Cloning into '/var/folders/lc/wjt2vfs50vsbd38_n8cp551h0000gn/T/d20161214-59866-xxnfgf'...

/opt/local/lib/ruby2.2/gems/2.2.0/gems/fastlane_core-0.60.0/lib/fastlane_core/command_executor.rb:65:in `wait': Interrupt
	from /opt/local/lib/ruby2.2/gems/2.2.0/gems/fastlane_core-0.60.0/lib/fastlane_core/command_executor.rb:65:in `ensure in block in execute'
	from /opt/local/lib/ruby2.2/gems/2.2.0/gems/fastlane_core-0.60.0/lib/fastlane_core/command_executor.rb:65:in `block in execute'
	from /opt/local/lib/ruby2.2/gems/2.2.0/gems/fastlane_core-0.60.0/lib/fastlane_core/command_executor.rb:46:in `spawn'
	from /opt/local/lib/ruby2.2/gems/2.2.0/gems/fastlane_core-0.60.0/lib/fastlane_core/command_executor.rb:46:in `execute'
	from /opt/local/lib/ruby2.2/gems/2.2.0/gems/match-0.11.1/lib/match/git_helper.rb:14:in `clone'
	from /opt/local/lib/ruby2.2/gems/2.2.0/gems/match-0.11.1/lib/match/runner.rb:13:in `run'
	from /opt/local/lib/ruby2.2/gems/2.2.0/gems/match-0.11.1/lib/match/commands_generator.rb:53:in `block (3 levels) in run'
	from /opt/local/lib/ruby2.2/gems/2.2.0/gems/commander-4.4.1/lib/commander/command.rb:178:in `call'
	from /opt/local/lib/ruby2.2/gems/2.2.0/gems/commander-4.4.1/lib/commander/command.rb:178:in `call'
	from /opt/local/lib/ruby2.2/gems/2.2.0/gems/commander-4.4.1/lib/commander/command.rb:153:in `run'
	from /opt/local/lib/ruby2.2/gems/2.2.0/gems/commander-4.4.1/lib/commander/runner.rb:444:in `run_active_command'
	from /opt/local/lib/ruby2.2/gems/2.2.0/gems/fastlane_core-0.60.0/lib/fastlane_core/ui/fastlane_runner.rb:38:in `run!'
	from /opt/local/lib/ruby2.2/gems/2.2.0/gems/commander-4.4.1/lib/commander/delegates.rb:15:in `run!'
	from /opt/local/lib/ruby2.2/gems/2.2.0/gems/match-0.11.1/lib/match/commands_generator.rb:119:in `run'
	from /opt/local/lib/ruby2.2/gems/2.2.0/gems/match-0.11.1/lib/match/commands_generator.rb:12:in `start'
	from /opt/local/lib/ruby2.2/gems/2.2.0/gems/match-0.11.1/bin/match:6:in `<top (required)>'
	from /opt/local/bin/match:23:in `load'
	from /opt/local/bin/match:23:in `<main>'

```

**Comment by martinmoserswiss** [comment]
@parleer did you find a solution for that problem?

**Comment by lacostej** [comment]
@parleer have you tried using `--verbose` option? If that didn't give more information, can you enable more `git clone` verbosity by using:
`GIT_SSH_COMMAND="ssh -vvv" fastlane ...`

**Comment by KrauseFx** [comment]
This issue should be resolved, please share more information about your setup, so we can help you 👍

**Comment by parleer** [comment]
Sorry I forgot to update this post... It is working now, but I don't remember why.

I believe all I needed to do was `ssh-add` my passphrase to ssh-agent... but in my comment I said that I had removed the passphrase altogether and it still didn't work. 

I know that I also reinstalled git, which may have had something to do with it as well.

FWIW, when I `ssh-add -D` and then run `fastlane certs` (which runs match), I get the exact same behavior. It hangs on "Cloning remote git repo..." That's expected behavior. 'ssh-add' fixes things.
